Tools/clinic/libclinic/converters.py

```diff
@@ -161,12 +161,19 @@ def parse_arg(self, argname: str, displayname: str, *, limited_capi: bool) -> st
         elif self.format_unit == 'B':
             return self.format_code("""
                 {{{{
-                    unsigned long ival = PyLong_AsUnsignedLongMask({argname});
-                    if (ival == (unsigned long)-1 && PyErr_Occurred()) {{{{
+                    Py_ssize_t _bytes = PyLong_AsNativeBytes({argname}, &{paramname}, sizeof(unsigned char),
```

Comment:
Can you factorize this code? It's copied 4 times. For example, write a function which returns this template, with an argument: the sizeof() argument.

Reply:
Actually, it is not the exactly the same. For `unsigned long` and `unsigned long long` it does additional check (historically).

But well, it can be factorized, and with recent changes in Argument Clinic, even more code can be factorized.